### ISSUE TITLE
Should the ubuntu package also recommend fuse-overlayfs ?

### ISSUE THREAD

**Issue by afbjorklund** [containers/podman#8560]
/kind feature

Currently the ubuntu packages uses "vfs" by default, for the rootless graph driver.

Maybe it should also recommend fuse-overlayfs, like the Fedora package does ?

### podman_2.2.0~2_amd64.deb
```
 Depends: libseccomp2, libdevmapper1.02.1, libgpgme11, catatonit, conmon (>= 2.0
.18~1), containers-common (>= 1.2.0~3), containernetworking-plugins (>= 0.8.6~1)
, runc | cri-o-runc, iptables, podman-plugins (>= 1.1.1~1)
 Recommends: crun, slirp4netns, uidmap, varlink
```

### podman-2.2.0-2.fc33.x86_64.rpm
```
--requires
(container-selinux if selinux-policy)
/bin/sh
config(podman) = 2:2.2.0-2.fc33
conmon >= 2:2.0.16-1
containernetworking-plugins >= 0.8.6-1
containers-common >= 1:1.2.0-9
iptables
nftables
oci-runtime
rtld(GNU_HASH)
--recommends
catatonit
crun >= 0.15-5
fuse-overlayfs >= 0.3-8
podman-plugins = 2:2.2.0-2.fc33
runc
slirp4netns >= 0.3.0-2
```

**Comment by mheon** [comment]
Are these the official Ubuntu/Debian packages, or the OBS packages? I'm assuming OBS from the 2.2.0 version?

@lsm5 PTAL

**Comment by rhatdan** [comment]
It should definitely recommend fuse-overlay.

**Comment by c-goes** [comment]
Maybe rootless Podman should default to overlay on Ubuntu/Debian if fuse-overlay isn't installed because the kernel supports it. Currently it uses VFS.

**Comment by afbjorklund** [comment]
> Are these the official Ubuntu/Debian packages, or the OBS packages? I'm assuming OBS from the 2.2.0 version?

This was the OBS package for Ubuntu 20.04 LTS, sorry (there are no `podman` packages for this release, only 20.10)

The `fuse-overlayfs` package is available in the regular Ubuntu repository, only side-effect was upgrading `fuse` (v2)

**Comment by zgoda** [comment]
Perhaps because `gvfs-fuse` package in Ubuntu 20.04 still depends on fuse2 and these two can not be installed at the same time.

**Comment by afbjorklund** [comment]
> Perhaps because `gvfs-fuse` package in Ubuntu 20.04 still depends on fuse2 and these two can not be installed at the same time.

Apparently it is happy enough as long as `libfuse2` is installed for the library, and then `fuse3` will fulfill the `fuse` dependency

**Comment by qhaas** [comment]
Ran into this while deploying to the latest LTS release of Ubuntu (20.04), can we get it added to [the documentation](https://podman.io/getting-started/installation#ubuntu).

**Comment by rhatdan** [comment]
Interested in opening a PR to fix the documentation?

**Comment by lsm5** [comment]
> > Are these the official Ubuntu/Debian packages, or the OBS packages? I'm assuming OBS from the 2.2.0 version?
> 
> This was the OBS package for Ubuntu 20.04 LTS, sorry (there are no `podman` packages for this release, only 20.10)
> 
> The `fuse-overlayfs` package is available in the regular Ubuntu repository, only side-effect was upgrading `fuse` (v2)
> 
> ```
> The following additional packages will be installed:
>   fuse3
> The following packages will be REMOVED:
>   fuse
> The following NEW packages will be installed:
>   fuse-overlayfs fuse3
> 0 upgraded, 2 newly installed, 1 to remove and 0 not upgraded.
> Need to get 62,7 kB of archives.
> After this operation, 87,0 kB of additional disk space will be used.
> ```
> 
> After this it works, but you do have to recreate storage (as usual).

So, help me understand, is it good enough if I have podman and podman-rootless packages `Recommends: fuse-overlayfs, fuse3` ? 

Also, I don't wanna make any changes to the podman package on the Kubic Stable repo until we have v3.0.0 released. Until then I'll be sending all the changes only to Kubic Testing. So Stable users would need to wait a while.

**Comment by afbjorklund** [comment]
> So, help me understand, is it good enough if I have podman and podman-rootless packages `Recommends: fuse-overlayfs, fuse3` ?

That is fine. The user workaround (for podman2 users), would be install "fuse-overlayfs" explicitly.

**Comment by lsm5** [comment]
> I don't understand the difference between podman and podman-rootless packages, though.
> 
> As far as I know, fuse-overlayfs and slirp4netns are only required when running rootless ?

That was on @mheon's suggestion where the regular package's dependency on containernetworking-plugins caused some issue which I don't remember off-hand.

**Comment by mheon** [comment]
Kubernetes provides its own CNI plugins, which caused our packages to conflict with Kubernetes. The new package's relaxed dependencies fixed that.

**Comment by lsm5** [comment]
Should be fixed with v3.0.0 on kubic stable. Please reopen if issue persists...
